addons/pos_hr/static/src/js/models.js

```diff
@@ -3,57 +3,26 @@ odoo.define('pos_hr.employees', function (require) {
 
 var models = require('point_of_sale.models');
 const Registries = require('point_of_sale.Registries');
-var pos_env = require('point_of_sale.env');
-
-models.load_models([{
-    model:  'hr.employee',
-    loaded: function(self, employees) {
-        if (self.config.module_pos_hr) {
-            self.employees = employees;
-            self.employee_by_id = {};
-            self.employees.forEach(function(employee) {
-                self.employee_by_id[employee.id] = employee;
-                var hasUser = self.users.some(function(user) {
-                    if (user.id === employee.user_id[0]) {
-                        employee.role = user.role;
-                        return true;
-                    }
-                    return false;
-                });
-                if (!hasUser) {
-                    employee.role = 'cashier';
-                }
-            });
-        }
-    }
-}]);
 
 Registries.PosModelRegistry.extend(models.PosGlobalState, (PosGlobalState) => {
 
 class PosHrPosModel extends PosGlobalState {
-    after_load_server_data() {
-        var self = this;
-        var employee_ids = _.map(self.employees, function(employee){return employee.id;});
-        var records = pos_env.services.rpc({
-            model: 'hr.employee',
-            method: 'get_barcodes_and_pin_hashed',
-            args: [employee_ids],
-        });
-        return records.then(function (employee_data) {
-            self.employees.forEach(function (employee) {
-                var data = _.findWhere(employee_data, {'id': employee.id});
-                if (data !== undefined){
-                    employee.barcode = data.barcode;
-                    employee.pin = data.pin;
-                }
-            });
-        }).then(async () => {
-            await super.after_load_server_data(...arguments);
+    async _processData(loadedData) {
+        await super._processData(...arguments);
+        if (this.config.module_pos_hr) {
+            this.employees = loadedData['hr.employee'];
```

Comment:
Basically, we might not need this. We can just replace it's use with `employee_by_id`.
Or, we create a getter called `getEmployees` that returns `Object.values(employee_by_id)`.

Reply:
added in todo list